drivers/sensor/b57861s0103a039/b57861s0103a039.c

```diff
@@ -0,0 +1,330 @@
+ * @y1: y value of point 1
+ * @x: the linear interpolant
+ */
+static inline int fixp_linear_interpolate(int x0, int y0, int x1, int y1, int x)
+{
+	if (y0 == y1 || x == x0) {
+		return y0;
+	}
+	if (x1 == x0 || x == x1) {
+		return y1;
+	}
+
+	return y0 + ((y1 - y0) * (x - x0) / (x1 - x0));
+}
+
+static int32_t get_temp_mc(struct b57861s0103a039_sensor_data *data, unsigned int ohm)
+{
+	int low, high;
+	int temp;
+	const struct ntc_type *table = data->comp_table;
+
+	lookup_comp(table, ohm, &low, &high);
+	/*
+	 * First multiplying the table temperatures with 1000 to get to
+	 * millicentigrades (which is what we want) and then interpolating
+	 * will give the best precision.
+	 */
+	temp = fixp_linear_interpolate(table->comp[low].ohm, table->comp[low].temp_c * 1000,
+				       table->comp[high].ohm, table->comp[high].temp_c * 1000, ohm);
+	return temp;
+}
+
+static uint32_t get_ohm_of_thermistor(const struct device *dev, int16_t raw_adc)
+{
+	const struct b57861s0103a039_sensor_config *cfg = dev->config;
+	uint32_t max_adc = (1 << (cfg->adc_channel.resolution - 1)) - 1;
+	uint32_t ohm, pullup_ohm, pulldown_ohm;
+
+	pullup_ohm = cfg->pullup_ohm;
+	pulldown_ohm = cfg->pulldown_ohm;
+
+	if (cfg->connection_type == NTC_CONNECTED_POSITIVE) {
+		ohm = pulldown_ohm * max_adc / (raw_adc - 1);
+	} else {
+		ohm = pullup_ohm * (raw_adc - 1) / max_adc;
+	}
+
+	return ohm;
+}
+
+static int b57861s0103a039_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct b57861s0103a039_sensor_data *data = dev->data;
+	const struct b57861s0103a039_sensor_config *cfg = dev->config;
+	int32_t val_mv;
+	int res;
+	struct adc_sequence sequence = {.options = NULL,
+					.buffer = &data->raw,
+					.buffer_size = sizeof(data->raw),
+					.calibrate = false};
+
+	k_mutex_lock(&data->mutex, K_FOREVER);
+
+	adc_sequence_init_dt(&cfg->adc_channel, &sequence);
+	res = adc_read(cfg->adc_channel.dev, &sequence);
+	val_mv = data->raw;
+	res = adc_raw_to_millivolts_dt(&cfg->adc_channel, &val_mv);
+	data->sample_val = val_mv;
+
+	k_mutex_unlock(&data->mutex);
+
+	return res;
+}
+
+static int b57861s0103a039_channel_get(const struct device *dev, enum sensor_channel chan,
+				       struct sensor_value *val)
+{
+	struct b57861s0103a039_sensor_data *data = dev->data;
+	uint32_t ohm;
+	int32_t temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		ohm = get_ohm_of_thermistor(dev, data->raw);
+		temp = get_temp_mc(dev->data, ohm);
+		val->val1 = temp / 1000;
+		val->val2 = (temp % 1000) * 1000;
+		LOG_ERR("ntc temp says %u", val->val1);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int b57861s0103a039_attr_set(const struct device *dev, enum sensor_channel chan,
+				    enum sensor_attribute attr, const struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
+
+static int b57861s0103a039_attr_get(const struct device *dev, enum sensor_channel chan,
+				    enum sensor_attribute attr, struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
```

Comment:
If these aren't needed you can just leave them out of the API, the sensor.h static inline functions will check these and return -ENOTSUP for you